### PR TITLE
started adding dagger

### PR DESCRIPTION
**Pull request by chayahaor**

### CODE REVIEW

File: src/main/java/ActivityModule.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import dagger.Provides;
+
+import javax.inject.Singleton;
+
+@Module
+public class ActivityModule {
+
+    @Singleton
+    @Provides
+    public ActivityService openWeatherMapService(GetActivityServiceFactory factory) {
```
Comment (andrewoid) [inline src/main/java/ActivityModule.java]: This is not an OpenWeatherMapService.

Be very careful when copying code.

File: src/main/java/ActivityPresenter.java [hunk 2]
```diff
@@ -30,13 +32,13 @@ public void onNext(Activity activity) {
         String nextActivity = activity.getNextActivity() + " can be done with "
                 + activity.getParticipants() + people;
         System.out.println(nextActivity);
-        view.setActivity(nextActivity);
-        view.setLink(activity.getLink());
+        viewProvider.get().setActivity(nextActivity);
+        viewProvider.get().setLink(activity.getLink());
```
Comment (andrewoid) [inline src/main/java/ActivityPresenter.java]: Call viewProvider.get() only once.

Comment (chayahaor) [reply src/main/java/ActivityPresenter.java]: How do I access the view without calling the viewProvider.get()?
When I tried removing the .get part, I got the error that viewProvider does not contain the method .setLink

Comment (andrewoid) [reply src/main/java/ActivityPresenter.java]: What I mean is don't repeat it twice.
Call it once and save the returned value.

File: src/main/java/ActivityPresenter.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
 public class ActivityPresenter {
     private Provider<ActivityFrame> viewProvider;
     private ActivityService model;
-
+    private ActivityFrame view;
```
Comment (andrewoid) [inline src/main/java/ActivityPresenter.java]: Why is this a member variable?

File: src/main/java/ActivityFrame.java [hunk 4]
```diff
@@ -112,7 +114,7 @@ public void showError() {
     }
 
     public static void main(String[] args) {
-        JFrame frame = new ActivityFrame();
+        ActivityFrame frame = DaggerActivityComponent.create().getActivityFrame();
         frame.setVisible(true);
         Image icon = Toolkit.getDefaultToolkit().getImage("icon.png");
         frame.setIconImage(icon);
```
Comment (andrewoid) [inline src/main/java/ActivityFrame.java]: You should do this within the ActivitFrame constructor.

File: src/main/java/GetActivityServiceFactory.java [hunk 5]
```diff
@@ -4,7 +4,15 @@
 import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
 import retrofit2.converter.gson.GsonConverterFactory;
 
+import javax.inject.Inject;
+
 public class GetActivityServiceFactory {
```
Comment (andrewoid) [inline src/main/java/GetActivityServiceFactory.java]: just `ActivityServiceFactory`

**Review by andrewoid — APPROVED** [review]

**Pull request merged**